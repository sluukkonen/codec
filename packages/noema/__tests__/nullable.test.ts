import { nullable, string } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("nullable", () => {
  test("should parse undefined or a value parsed by the underlying codec", () => {
    expectParseSuccess(nullable(string), null)
    expectParseSuccess(nullable(string), "")
  })

  test("should fail to parse other values", () => {
    expectParseFailure(nullable(string), 1, [
      { code: "invalid_string", path: [] },
    ])
  })
})
