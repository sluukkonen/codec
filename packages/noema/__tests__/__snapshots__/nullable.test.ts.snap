// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nullable should fail to parse other values 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "string",
    "path": [],
    "received": "number",
  },
]
`;
